src/xunit.performance.api/XunitPerformanceHarness.cs

```diff
@@ -32,53 +39,72 @@ public XunitPerformanceHarness(string[] args)
             Configuration.FileLogPath = $"{Configuration.RunId}.csv"; // TODO: Conditionally set this based on whether we want a csv file written.
         }
 
+        /// <summary>
+        /// Gets the path for the output directory.
+        /// </summary>
         public string OutputDirectory { get; }
 
+        /// <summary>
+        /// Gets a collection of the type names of the test classes to run.
+        /// </summary>
+        /// <remarks>Returns an empty collection if the type names were not specified.</remarks>
         public IEnumerable<string> TypeNames => _typeNames.AsReadOnly();
 
+        /// <summary>
+        /// 
+        /// </summary>
         public BenchmarkConfiguration Configuration => BenchmarkConfiguration.Instance;
 
+        /// <summary>
+        /// Run the xUnit tests tagged with the [<see cref="BenchmarkAttribute"/>] attribute.
+        /// </summary>
+        /// <param name="assemblyFileName">Path to the assembly that contains the xUnit performance tests.</param>
         public void RunBenchmarks(string assemblyFileName)
         {
             if (string.IsNullOrEmpty(assemblyFileName))
                 throw new ArgumentNullException(nameof(assemblyFileName));
             if (!File.Exists(assemblyFileName))
                 throw new FileNotFoundException(assemblyFileName);
 
-            Action<string> xUnitAction = (assemblyPath) => { XunitRunner.Run(assemblyPath, _typeNames); };
+            void xUnitAction(string assemblyPath)
+            {
+                XunitRunner.Run(assemblyPath, _typeNames);
+            }
+
             var xUnitPerformanceSessionData = new XUnitPerformanceSessionData {
                 AssemblyFileName = assemblyFileName,
-                CollectOutputFilesCallback = (fileName) => {
-                    // FIXME: This will need safe guards when the client calls RunBenchmarks in different threads.
-                    _outputFiles.Add(fileName);
-                    WriteInfoLine($"File saved to: \"{fileName}\"");
-                },
+                CollectOutputFilesCallback = LogFileSaved,
```

Comment:
It looks like we no longer add fileName to _outputFiles. This had a FIXME attached to it, is this fixed? I did not see anything else in the diff that obviously fixed this.

Reply:
It is not used, so I removed it. We only use this function to display generated files.

Reply:
Alright, sounds good to me.